src/libraries/System.Memory/tests/Span/StringSearchValues.cs

```diff
@@ -0,0 +1,450 @@
+
+                if (comparisonType == StringComparison.OrdinalIgnoreCase)
+                {
+                    AssertIndexOfAnyAndFriends(new[] { differentCase }, 0, -1, 0, -1);
+                    AssertIndexOfAnyAndFriends(new[] { differentCase, differentCase }, 0, -1, 1, -1);
+                    AssertIndexOfAnyAndFriends(new[] { differentCase, ValueNotInSet }, 0, 1, 0, 1);
+                    AssertIndexOfAnyAndFriends(new[] { differentCase, ValueNotInSet, ValueNotInSet }, 0, 1, 0, 2);
+                    AssertIndexOfAnyAndFriends(new[] { ValueNotInSet, differentCase }, 1, 0, 1, 0);
+                    AssertIndexOfAnyAndFriends(new[] { ValueNotInSet, ValueNotInSet, differentCase }, 2, 0, 2, 1);
+                    AssertIndexOfAnyAndFriends(new[] { ValueNotInSet, differentCase, ValueNotInSet }, 1, 0, 1, 2);
+                    AssertIndexOfAnyAndFriends(new[] { differentCase, ValueNotInSet, differentCase }, 0, 1, 2, 1);
+                }
+                else
+                {
+                    AssertIndexOfAnyAndFriends(new[] { differentCase }, -1, 0, -1, 0);
+                    AssertIndexOfAnyAndFriends(new[] { differentCase, differentCase }, -1, 0, -1, 1);
+                    AssertIndexOfAnyAndFriends(new[] { differentCase, ValueNotInSet }, -1, 0, -1, 1);
+                    AssertIndexOfAnyAndFriends(new[] { ValueNotInSet, differentCase }, -1, 0, -1, 1);
+                    AssertIndexOfAnyAndFriends(new[] { differentCase, ValueNotInSet, ValueNotInSet }, -1, 0, -1, 2);
+                }
+            }
+
+            void AssertIndexOfAnyAndFriends(Span<string> values, int any, int anyExcept, int last, int lastExcept)
+            {
+                Assert.Equal(any >= 0, last >= 0);
+                Assert.Equal(anyExcept >= 0, lastExcept >= 0);
+
+                Assert.Equal(any, values.IndexOfAny(stringValues));
+                Assert.Equal(any, ((ReadOnlySpan<string>)values).IndexOfAny(stringValues));
+                Assert.Equal(anyExcept, values.IndexOfAnyExcept(stringValues));
+                Assert.Equal(anyExcept, ((ReadOnlySpan<string>)values).IndexOfAnyExcept(stringValues));
+                Assert.Equal(last, values.LastIndexOfAny(stringValues));
+                Assert.Equal(last, ((ReadOnlySpan<string>)values).LastIndexOfAny(stringValues));
+                Assert.Equal(lastExcept, values.LastIndexOfAnyExcept(stringValues));
+                Assert.Equal(lastExcept, ((ReadOnlySpan<string>)values).LastIndexOfAnyExcept(stringValues));
+
+                Assert.Equal(any >= 0, values.ContainsAny(stringValues));
+                Assert.Equal(any >= 0, ((ReadOnlySpan<string>)values).ContainsAny(stringValues));
+                Assert.Equal(anyExcept >= 0, values.ContainsAnyExcept(stringValues));
+                Assert.Equal(anyExcept >= 0, ((ReadOnlySpan<string>)values).ContainsAnyExcept(stringValues));
+            }
+        }
+
+        [Theory]
+        // Sets with empty values
```

Comment:
are any of these worth borrowing?
https://github.com/BurntSushi/aho-corasick/blob/master/src/tests.rs#L91

Reply:
I added a few simple cases, but I think we have decent test coverage otherwise.